ui/src/pages/management/components/UserRolesTable/index.tsx

```diff
@@ -0,0 +1,194 @@
+import { Table, Tag, Button, message, Popconfirm } from "antd";
+import { DeleteOutlined } from "@ant-design/icons";
+import { ColumnsType } from "antd/lib/table";
+import dayjs from "dayjs";
+import { UserRole } from "../../../../models/model";
+import { listUserRole, deleteUserRole } from "../../../../api";
+
+export interface UserRolesTableProps {}
+
+export interface UserRolesTableInstance {
+  onSearch?: (values: any) => void;
+}
+
+export interface SearchModel {
+  scope?: string;
+  roleName?: string;
+}
+
+function UserRolesTable(props: UserRolesTableProps, ref: any) {
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const [tableData, setTableData] = useState<UserRole[]>([]);
+
+  const searchRef = useRef<SearchModel>();
+
+  const fetchData = useCallback(async () => {
+    setLoading(true);
+    try {
+      let result = await listUserRole();
+      if (searchRef.current) {
+        const { scope, roleName } = searchRef.current;
+        result = result.filter((item) => {
+          let value = true;
+          if (scope) {
+            value = item.scope.includes(scope.toLocaleLowerCase());
+          }
+          if (value && roleName) {
+            value = item.roleName === roleName;
+          }
+          return value;
+        });
+      }
+      result.sort((a: UserRole, b: UserRole) => {
+        return dayjs(b.createTime).diff(
+          dayjs(a.createTime),
+          "milliseconds",
+          true
+        );
+      });
+
+      setTableData(result);
+    } catch {
+      //
+    } finally {
+      setLoading(false);
+    }
+  }, []);
+
+  const onDelete = async (row: UserRole) => {
+    console.log(
+      `The [${row.roleName}] Role of [${row.userName}] user role delete request is sent.`
+    );
+    try {
+      await deleteUserRole(row);
+      message.success(`Role ${row.roleName} of user ${row.userName} deleted`);
+      fetchData();
+    } catch {
+      message.error("Failed to delete userrole.");
+    }
+  };
+
+  const columns: ColumnsType<UserRole> = [
+    {
+      title: "Scope (Project / Global)",
+      dataIndex: "scope",
+      ellipsis: true,
+    },
+    {
+      title: "Role",
+      dataIndex: "roleName",
+      width: 120,
+    },
+    {
+      title: "User",
+      dataIndex: "userName",
+      ellipsis: true,
+    },
+    {
+      title: "Permissions",
+      dataIndex: "access",
+      width: 240,
+      render: (col: any[]) => {
+        return col.map((tag) => {
+          let color = tag.length > 5 ? "red" : "green";
+          if (tag === "write") color = "blue";
+          return (
+            <Tag color={color} key={tag}>
+              {tag.toUpperCase()}
+            </Tag>
+          );
+        });
+      },
+    },
+    {
+      title: "Reason",
+      dataIndex: "createReason",
+      ellipsis: true,
+    },
+    {
+      title: "Create By",
+      dataIndex: "createBy",
+      width: 200,
+      ellipsis: true,
+    },
+    {
+      title: "Create Time (Local)",
```

Comment:
Create Time (Local) => Create Time

Reply:
Boli has made a localization to the `CreateTime` based on browser timezone. Thus it could be different from the utc record in the Database and will cause misunderstanding.
I asked him to modify the title to indicate user that it is a local time. Or maybe we could just add a notification above?

Reply:
I meant by default time displayed in UI is always local time, then we don't need to add a local in column title.